queueing/inserter.py

```diff
@@ -25,40 +31,58 @@ def __init__(self, client, database_name, collection_name, unique):
 
         logging.info("New Inserter created for {}.".format(self.identifier))
 
-    def __insert_items(self):
-        response = self.__client.insert_many_to_collection(
-            self.database_name, self.collection_name, self.__items
+    def __handle_insertion_error(self, response, items):
+        logging.warning(
+            "The exception {} occured during the insertion of {} items in {}.".format(
+                response.message, len(items), self.identifier
+            )
         )
-        logging.debug(
-            "{} documents inserted in {}.".format(len(self.__items), self.identifier)
+        for item in items:
+            if response.need_upsert:
+                item["need_upsert"] = "True"
+            kafka_producer.send(self.topic, value=item)
+
+    def __insert_items(self, reason=""):
+        items = [copy(item["payload"]) for item in self.__items]
```

Comment:
Why do we copy the payload? Would it be possible to use the original item to reduce overhead and memory use?

Reply:
Done, now we erase the "_id" field if exists